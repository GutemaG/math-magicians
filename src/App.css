
.btns {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  width: 550px;
  cursor: pointer;
}

.calculator-input {
  width: 510px;
  padding: 20px;
  background-color: #ccc;
  border: none;
  direction: rtl;
}

.calculator-input:focus {
  border: none;
  outline: none;
}

.btn {
  border: 1px solid #ccc;
  padding: 20px;
  text-align: center;
}

.operator {
  background-color: orange;
}

.zero {
  grid-column: 1/ span 2;
}
