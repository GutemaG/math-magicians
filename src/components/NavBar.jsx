import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import './styles/NavBar.css';

const NavBar = () => (
  <nav className="nav">
    <Link to="/" className="logo">Math Magicians</Link>
    <ul className="nav-items">
      <li className="nav-item">
        <NavLink to="/">Home</NavLink>
      </li>
      <li className="nav-item">
        <NavLink to="/calculator">Calculator</NavLink>
      </li>
      <li className="nav-item">
        <NavLink to="/quote">Quote</NavLink>
      </li>
    </ul>
  </nav>
);
export default NavBar;
