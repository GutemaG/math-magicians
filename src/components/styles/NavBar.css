.nav {
  background: #ccc;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.logo {
  text-transform: uppercase;
  padding: 1rem;
  font-size: 1.5rem;
}

.nav-items {
  display: flex;
  gap: 10px;
}

.nav-items .nav-item a {
  padding: 1rem;
}

.nav-item a:hover {
  background-color: #ddd;
  border-radius: 10px;
  text-decoration: underline;
}

.active {
  font-weight: bold;
  text-decoration: underline;
}
