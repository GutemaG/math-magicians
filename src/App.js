import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Calculator from './components/Calculator';
import Home from './components/Home';
import NavBar from './components/NavBar';
import Quote from './components/Quote';

const App = () => {
  const style = {
    padding: '2rem',
  };
  return (
    <>
      <NavBar />
      <div style={style}>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/calculator" element={<Calculator />} />
          <Route path="/quote" element={<Quote />} />
        </Routes>
      </div>
    </>
  );
};

export default App;
