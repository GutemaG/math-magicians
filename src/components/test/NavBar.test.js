import {
  fireEvent,
  render,
  screen,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from '../../App';
import NavBar from '../NavBar';

describe('Testing Component render correctly', () => {
  it('NavBar rendered correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('NavBar interaction', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('Home'));
    const welcomeMsg = screen.getByText('Welcome to our Page!');
    expect(welcomeMsg).toBeInTheDocument();
  });
});
