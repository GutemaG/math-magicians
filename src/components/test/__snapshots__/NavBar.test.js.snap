// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Component render correctly NavBar rendered correctly 1`] = `
<nav
  className="nav"
>
  <a
    className="logo"
    href="/"
    onClick={[Function]}
  >
    Math Magicians
  </a>
  <ul
    className="nav-items"
  >
    <li
      className="nav-item"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
